core/tests/puppeteer-acceptance-tests/user-utilities/question-admin-utils.js

```diff
@@ -0,0 +1,163 @@
+   */
+  async addSubmitQuestionRights(username) {
+    await this.type(addContributorUsernameInput, username);
+    await this.select(
+      addContributonRightsCategorySelect, submitQuestionRightValue);
+    await this.clickOn(addContributionRightsSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+  }
+
+  /**
+   * Function to display contribution rights by user.
+   * @param {string} username - the username of the user to view.
+   */
+  async viewContributionRightsForUser(username) {
+    await this.select(viewContributorFilterMethodSelect, usernameMethodValue);
+    await this.type(viewContributerUsernameInput, username);
+    await this.clickOn(viewContributorSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+  }
+
+  /**
+   * Function to display reviewing questions rights by role.
+   */
+  async viewContributorReviewQuestionRightsByRole() {
+    await this.select(viewContributorFilterMethodSelect, roleMethodValue);
+    await this.select(viewContributorCategorySelect, reviewQuestionRightValue);
+    await this.clickOn(viewContributorSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+  }
+
+  /**
+   * Function to check if the username is displayed as one of the users
+   * with rights of reviewing question.
+   * @param {string} username - the username of the user to check.
+   */
+  async expectDisplayedUsersToContainReviewQuestionRights(username) {
+    await this.page.waitForSelector(viewContributorReviewQuestionsResult);
+    const displayedUsername = await this.page.$eval(
+      viewContributorReviewQuestionsResult,
+      element => element.innerText);
+    if (displayedUsername.includes('Not-allowed')) {
+      throw new Error(
+        `${username} does not have rights for reviewing questions!`);
+    } else {
+      showMessage(
+        `${username} has rights for reviewing questions!`);
+    }
+  }
+
+  /**
+   * Function to check if the user is displayed as a question reviewer.
+   * @param {string} username - the user expected to be displayed.
+   */
+  async expectUserToBeDisplayed(username) {
+    await this.page.waitForSelector(viewReviewQuestionRoleUserResult);
+    const displayedUsers = await this.page.$eval(
+      viewReviewQuestionRoleUserResult,
+      element => element.innerText
+    );
+    if (!displayedUsers.includes(username)) {
+      throw new Error(
+        `${username} does not have rights for reviewing questions!`);
+    }
+  }
+
+  /**
+   * Function to check that there are no question reviewer.
+   * @param {string} usename - the user expected to not be displayed.
+   */
+  async expectUserToNotBeDisplayed(username) {
+    await this.page.waitForSelector(viewReviewQuestionRoleUserResult);
+    const displayedUsers = await this.page.$eval(
+      viewReviewQuestionRoleUserResult,
+      element => element.innerText
+    );
+    if (displayedUsers.includes(username)) {
+      throw new Error(
+        `${username} still has the right to review question!`);
```

Comment:
This function could also be used as a pre-check (i.e. to ensure that at the start of the test, the user doesn't have reviewing rights), but in that scenario, the word "still" wouldn't make sense. To accommodate this use case, consider re-wording the message to `${username} has the right to review questions!`

Reply:
Thanks for pointing it out, It's fixed now.